js/apps/admin-ui/src/context/RealmsContext.tsx

```diff
@@ -5,25 +5,31 @@ import { createNamedContext, useRequiredContext } from "ui-shared";
 import { keycloak } from "../keycloak";
 import { useFetch } from "../utils/useFetch";
 import { fetchAdminUI } from "./auth/admin-ui-endpoint";
+import { sortBy } from "lodash-es";
 
 type RealmsContextProps = {
   /** A list of all the realms. */
-  realms: string[];
+  realms: RealmNameRepresentation[];
   /** Refreshes the realms with the latest information. */
   refresh: () => Promise<void>;
 };
 
+export interface RealmNameRepresentation {
+  name: string;
+  displayName: string | undefined;
+}
+
 export const RealmsContext = createNamedContext<RealmsContextProps | undefined>(
   "RealmsContext",
   undefined,
 );
 
 export const RealmsProvider = ({ children }: PropsWithChildren) => {
-  const [realms, setRealms] = useState<string[]>([]);
+  const [realms, setRealms] = useState<RealmNameRepresentation[]>([]);
   const [refreshCount, setRefreshCount] = useState(0);
 
-  function updateRealms(realms: string[]) {
-    setRealms(realms.sort());
+  function updateRealms(realms: RealmNameRepresentation[]) {
+    setRealms(sortBy(realms, "displayName", "name"));
```

Comment:
We're trying to reduce our dependency on Lodash where possible, as it is no longer actively maintained. This will also need to be sorted in a localized manner, for which I would recommend using [`useLocaleSort()`](https://github.com/keycloak/keycloak/blob/53a9eb1d429e13f363512ae04ae1646221dc7a3b/js/apps/a.dmin-ui/src/utils/useLocaleSort.ts#L5).

Reply:
I just tried implementing this and realized that the useLocalSort depends on the WhoAmIContext which is not yet available in the RealmsContext :/

![image](https://github.com/keycloak/keycloak/assets/1793007/e965d15e-e410-428f-a741-1ca9c68473dc)

I would have to rearrange the contexts like this:

![image](https://github.com/keycloak/keycloak/assets/1793007/b87af820-c9b6-4e9d-900e-7b2dac32b45d)

Reply:
I see you switched them around, I think that is a good solution :+1: